### DISCUSSION TITLE
How should we finish a bit signaling when the WIP is already activated?

### DISCUSSION THREAD

**Discussion by lrubiorod** [witnet/witnet-rust#2052] [Q&A]

**Comment by drcpu-github** [comment]
Personally, I think we should leave the signaling bit active until either one of these things

1. Until there is a conflict in voting bits because too many TAPI's have been used.
2. A new release is created which cements the activated rules into code (at which point it makes less sense to keep it active).
3. A new TAPI is created and up for voting.

I would not reset it immediately after a succesful TAPI because it offers useful insights into how many nodes are actually on board with the changes.

**Reply by aesedepece** [reply]
As per WIP-0014, point 2 should happen as soon as a new release is made. Guards that protect business logic from being executed unless a WIP is activated should be replaced by a simple epoch check.

**Reply by drcpu-github** [reply]
Yes, I know, just pointing out that that would also be a potential time for when to stop the signaling (though it may not be the best one).
